library/src/commonMain/kotlin/com/chrisjenx/yakcov/ValueValidator.kt

```diff
@@ -192,27 +194,35 @@ abstract class ValueValidator<V, R>(
      *  or if loosing focus when [validateOnFocusLost] is `true`.
      *  When you call [validate] it will start validating showing errors if present.
      */
-    @Composable
     fun Modifier.validationConfig(
         validateOnFocusLost: Boolean = false,
         shakeOnInvalid: Boolean = false,
         showErrorOnInteraction: Boolean = true,
-    ): Modifier {
+    ): Modifier = composed {
         // Set scope if shaking is enabled
         shakeOnInvalidScope = if (shakeOnInvalid) rememberCoroutineScope() else null
         // Track if we should show error on user input
         showErrorOnUserInput = showErrorOnInteraction
-        // Track focus
-        var hadFocus by mutableStateOf(false)
-        return this
-            .onFocusChanged { focusState ->
-                if (focusState.hasFocus) hadFocus = true
+        val validateOnFocusLostModifier = if (validateOnFocusLost) {
+            var hadFocus by remember { mutableStateOf(false) }
+            Modifier.onFocusChanged { focusState ->
                 // Don't shake on loss of focus, as we want to just show the error
-                if (validateOnFocusLost && !focusState.isFocused && hadFocus) {
-                    onValueChange(value = null)
+                if (validateOnFocusLost && hadFocus && !focusState.hasFocus) {
+                    validate(value = null, shake = false, shouldShowError = true)
                 }
+                hadFocus = focusState.hasFocus
             }
-            .shakable(shakingState)
+        } else {
+            Modifier
+        }
+        val shakeOnInvalidModifier = if (shakeOnInvalid) {
```

Comment:
could probably be simplified to something like:

```
this.let {
   if(shakeOnInvalid) then(Modifier.shakable(shakingState) else it
 }
 ```

Reply:
Sure, want me to inline `validateOnFocusLostModifier` as well?

Reply:
Also this can be

```kotlin
this.then(if (shakeOnInvalid) Modifier.shakable(shakingState) else Modifier)
```